src/main/java/se/liu/ida/hefquin/engine/data/mappings/impl/EntityMappingImpl.java

```diff
@@ -25,20 +27,86 @@ public EntityMappingImpl( final Graph mappingDescription ) {
 	}
 
 	protected void parseMappingDescription( final Graph mappingDescription ) {
-		// TODO: populate both g2lMap and l2gMap based on the owl:sameAs
+		// populate both g2lMap and l2gMap based on the owl:sameAs
 		// statements in the given RDF graph; assume the the subject of
 		// any owl:sameAs statement is a global URI and the object is the
 		// corresponding local URI
 		final Iterator<Triple> it = mappingDescription.find(Node.ANY, OWL2.sameAs.asNode(), Node.ANY);
 		while ( it.hasNext() ) {
-			// TODO ...
+			if(g2lMap.containsKey(it.next().getSubject())) {
+				// There is already at least one local URI for this global URI,
+				// We don't want to override it, so we'll add the new one to the set.
+				final Set<Node> locals = g2lMap.get(it.next().getSubject());
+				locals.add(it.next().getObject());
+				g2lMap.replace(it.next().getSubject(),locals);
```

Comment:
This should not be necessary. The program variable `locals` is a direct reference to the Java object that is also referenced from within the map. Manipulating that Java object via `locals.add` is enough. You don't need to put the reference to that Java object again into the map.